src/components/date-picker/z-range-picker/index.tsx

```diff
@@ -0,0 +1,628 @@
+        document.activeElement.classList.contains(`${this.rangepickerid}-2`)
+      ) {
+        !isCalendarOpened && currentInputState.picker.open();
+      }
+
+      let isPrevArrowEntered = document.activeElement.classList.contains(
+        "flatpickr-prev-month"
+      );
+      let isNextArrowEntered = document.activeElement.classList.contains(
+        "flatpickr-next-month"
+      );
+      let arrowPressed = isPrevArrowEntered || isNextArrowEntered;
+
+      arrowPressed && ev.key === " " && ev.preventDefault();
+
+      isPrevArrowEntered && currentInputState.picker.changeMonth(-1);
+      isNextArrowEntered && currentInputState.picker.changeMonth(1);
+    }
+  }
+
+  //Set background color before or after first selected date, before selecting the second one
+  @Listen("mouseover")
+  onMouseOver(e: CustomEvent) {
+    let currentInputState =
+      this.activeInput === RangePickerActiveInput.startInput
+        ? { picker: this.flatpickrInstance, index: 0 }
+        : { picker: this.flatpickrInstance2, index: 1 };
+
+    this.setRangeHoverStyle(
+      e,
+      currentInputState.picker,
+      this.element.getElementsByClassName("flatpickr-calendar")[
+        currentInputState.index
+      ]
+    );
+  }
+
+  componentWillLoad() {
+    let customToggle = this.element.querySelector("[slot=toggle]");
+    this.hasChildren = !!customToggle;
+    this.hasChildren && customToggle.setAttribute("data-toggle", "data-toggle");
+  }
+
+  componentDidLoad() {
+    let config = {
+      enableTime: this.mode === ZDatePickerMode.dateTime,
+      locale: Italian,
+      allowInput: true,
+      dateFormat:
+        this.mode === ZDatePickerMode.dateTime ? "d-m-Y - H:i" : "d-m-Y",
+      ariaDateFormat: "d F Y",
+      minuteIncrement: 1,
+      time_24hr: true,
+      onChange: (selectedDates, _dateStr, instance) => {
+        this.onDateSelect(selectedDates, instance);
+      },
+      onOpen: () => {
+        this.setAriaOptions();
+        this.setFlatpickrPosition();
+        this.setRangeStyle();
+        this.getFocusedInput();
+        this.getCurrentMonth();
+      },
+      onYearChange: () => {
+        this.setRangeStyle();
+      },
+      onMonthChange: () => {
+        this.setRangeStyle();
+      },
+      onKeyDown: () => {
+        this.setAriaOptions();
+      },
+      wrap: true,
+    };
+
+    this.flatpickrInstance = flatpickr(`.${this.rangepickerid}-container`, {
+      ...config,
+      mode: "multiple",
+      appendTo: this.element.children[0].children[0].children[0] as HTMLElement,
+    });
+    this.flatpickrInstance2 = flatpickr(`.${this.rangepickerid}-container-2`, {
+      ...config,
+      mode: "multiple",
+      appendTo: this.element.children[0].children[0].children[1] as HTMLElement,
+    });
+
+    this.element.querySelectorAll(".flatpickr-weekday").forEach((element) => {
+      element.innerHTML = element.innerHTML.trim().charAt(0);
+    });
+  }
+
+  AddClassToInput() {
```

Comment:
come convenzione mettiamo sempre il nome delle funzioni in camelCase
```suggestion
  addClassToInput() {
```